lib/trip/event.rb

```diff
@@ -214,6 +216,25 @@ def line?
   end
   # @endgroup
 
+  ##
+  # @return [Hash]
+  #  Returns a Hash object that can be serialized to JSON.
+  def as_json
+    {
+      "event" => name.to_s, "path" => path,
+      "lineno" => lineno, "method_id" => method_id.to_s,
+      "method_type" => (Module === @tp[:self]) ? "singleton_method" : "instance_method",
```

Comment:
Add `method_type` method to `Trip::Event`. Return nil when event is not a call or return.

Reply:
https://github.com/0x1eef/trip.rb/commit/db2dc99f59167e9d2823f41cf5c6078ca721fadf